File: career-connect-job-portal/job-portal-frontend/src/services/companyService.js
import API from "../api/axios";

export const getCompanies = () => API.get("/company");

export const addCompany = (company) => API.post("/company", company);

export const updateCompany = (id, company) =>
    API.put(`/company/${id}`, company);

export const deleteCompany = (id) =>
    API.delete(`/company/${id}`);

export const getCompanyById = (id) =>
    API.get(`/company/${id}`);

File: career-connect-job-portal/jobportal/src/main/java/com/onkar/jobportal/service/CompanyService.java
package com.onkar.jobportal.service;

import java.util.List;

import com.onkar.jobportal.dto.ApiResponse;
import com.onkar.jobportal.dto.CompanyRequest;
import com.onkar.jobportal.entity.Company;

public interface CompanyService {

	ApiResponse addCompany(Long recruiterId, CompanyRequest request);

    List<Company> getAllCompanies();

    Company getCompanyById(Long id);

    ApiResponse updateCompany(Long id, CompanyRequest request);

    ApiResponse deleteCompany(Long id);

}